pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java

```diff
@@ -2383,42 +2383,55 @@ private PositionImpl calculatePositionAckSet(boolean isExcluded, int batchSize,
+                .thenCompose(__ -> {
+                    CompletableFuture<Void> ret;
+                    if (topicName.isGlobal()) {
+                        ret = validateGlobalNamespaceOwnershipAsync(namespaceName);
+                    } else {
+                        ret = CompletableFuture.completedFuture(null);
+                    }
+                    return ret.thenCompose(ignore -> {
+                        return getTopicReferenceAsync(topicName)
+                                .thenAccept(topic -> {
+                                    ManagedLedgerImpl ledger =
+                                            (ManagedLedgerImpl) ((PersistentTopic) topic).getManagedLedger();
+                                    ledger.asyncReadEntry(new PositionImpl(ledgerId, entryId),
+                                            new AsyncCallbacks.ReadEntryCallback() {
+                                                @Override
+                                                public void readEntryFailed(ManagedLedgerException exception,
+                                                                            Object ctx) {
+                                                    asyncResponse.resume(new RestException(exception));
+                                                }
 
-                @Override
-                public void readEntryComplete(Entry entry, Object ctx) {
-                    try {
-                        asyncResponse.resume(generateResponseWithEntry(entry));
-                    } catch (IOException exception) {
-                        asyncResponse.resume(new RestException(exception));
-                    } finally {
-                        if (entry != null) {
-                            entry.release();
-                        }
+                                                @Override
+                                                public void readEntryComplete(Entry entry, Object ctx) {
+                                                    try {
+                                                        asyncResponse.resume(generateResponseWithEntry(entry));
+                                                    } catch (IOException exception) {
+                                                        asyncResponse.resume(new RestException(exception));
+                                                    } finally {
+                                                        if (entry != null) {
+                                                            entry.release();
+                                                        }
+                                                    }
+                                                }
+                                            }, null);
+                                });
+                    });
+                }).exceptionally(ex -> {
+                    Throwable cause = ex.getCause();
+                    if (cause instanceof NullPointerException) {
```

Comment:
it's weird. why we catch NPE here ? 

I mean, if somewhere gets NPE, we need to check there. Shouldn't it be catching exceptions?

Reply:
Yes, maybe we will fix this in another new pr.  If fixed here, will be confused.

Reply:
Agree with you.

Reply:
Please drop this check.
It is the right time to get rid of this tech debt.

If someone sees the NPE we can fix it.
If we keep this now I bet no one will take care.

It is also possible that that NPE was due to some old bug


I would like to see you dropping this line

Reply:
Done, This is involved by [6331](https://github.com/apache/pulsar/pull/6331). Seem that we can drop this line directly.